syntax/go.vim

```diff
@@ -240,6 +246,13 @@ else
                         \ contains=ALLBUT,goParen,goBlock,goFunction,goTypeName,goReceiverType,goReceiverVar
 endif
 
+if s:fold_comment
+	" changed this to comments instead of declaration_comment
+  syn region    goComment           start="/\*" end="\*/" contains=@goCommentGroup,@Spell fold
+  syn match     goComment           "\v(^\s*//.*\n)+" contains=goGenerate,@goCommentGroup,@Spell fold
+endif
```

Comment:
Looks good and works well in my testing, but could you move this next to the regular `goComment` lines on line 168? I think that makes more sense and would make it easier to maintain later on.

Would also be nice if you could remove the extra blank lines you added :-)